Sources/Views/MessagesCollectionView.swift

```diff
@@ -85,19 +85,28 @@ open class MessagesCollectionView: UICollectionView {
     
     private func setupGestureRecognizers() {
         let tapGesture = UITapGestureRecognizer(target: self, action: #selector(handleTapGesture(_:)))
-        tapGesture.delaysTouchesBegan = true
+        tapGesture.delegate = self
+
+        tapGesture.require(toFail: self.panGestureRecognizer)
+        if let pinchGestureRecognizer = self.pinchGestureRecognizer {
+            tapGesture.require(toFail: pinchGestureRecognizer)
```

Comment:
I did not add `directionalPressGestureRecognizer` since it crashes on iOS 12, and it is disabled by default anyway.